lib/exprengine.cpp

```diff
@@ -303,7 +303,7 @@ namespace {
             }
         }
 
-        void report(std::ostream &out, const Scope *functionScope) {
+        void report(std::ostream &out, const Scope *functionScope) const {
```

Comment:
I feel this is a 50/50 if I like it or not. This method does not affect its "this" object but there are other side effects.